marlowe-runtime/src/Language/Marlowe/Runtime/Transaction/Api.hs

```diff
@@ -0,0 +1,322 @@
+{-# LANGUAGE EmptyDataDeriving #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE TypeOperators #-}
+
+module Language.Marlowe.Runtime.Transaction.Api
+  where
+
+import Cardano.Api (AsType(..), ScriptDataSupportedInEra(..), Tx, TxBody, deserialiseFromCBOR, serialiseToCBOR)
+import qualified Data.Aeson as Aeson
+import Data.Binary (Binary, Get, get, getWord8, put)
+import Data.Binary.Put (Put, putWord8)
+import Data.ByteString (ByteString)
+import Data.Map (Map)
+import Data.Set (Set)
+import Data.Some (Some(..))
+import Data.Time (UTCTime)
+import Data.Type.Equality (type (:~:)(Refl))
+import Data.Void (Void, absurd)
+import GHC.Generics (Generic)
+import Language.Marlowe.Runtime.ChainSync.Api (Address, BlockHeader, TokenName, TxId, getUTCTime, putUTCTime)
+import Language.Marlowe.Runtime.Core.Api
+import Language.Marlowe.Runtime.Transaction.Constraints (UnsolvableConstraintsError)
+import Network.Protocol.Job.Types (Command(..), SomeTag(..))
+
+-- | The low-level runtime API for building and submitting transactions.
+data MarloweTxCommand status err result where
+  -- | Construct a transaction that starts a new Marlowe contract. The
+  -- resulting, unsigned transaction can be signed via the cardano API or a
+  -- wallet provider. When signed, the 'Submit' command can be used to submit
+  -- the transaction to the attached Cardano node.
+  Create
```

Comment:
We want to probably add `StakeAddressReference` here.

Reply:
Good idea, added